app/code/Magento/InventoryIndexer/Test/Integration/Plugin/InventoryApi/InvalidateAfterSourceSetEnabledPluginTest.php

```diff
@@ -0,0 +1,186 @@
+    {
+        $this->indexerInvalidationBase($sourceCode, $enable, $expectedValid);
+    }
+
+    /**
+     * Tests Source enabling and disabling when no Stocks are connected to current Source.
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     *
+     * @dataProvider IndexerInvalidationNoStockLinksDataProvider
+     * @param string $sourceCode
+     * @param bool $enable
+     * @param bool $expectedValid
+     */
+    public function testIndexerInvalidationNoStockLinks(string $sourceCode, bool $enable, bool $expectedValid)
+    {
+        $this->indexerInvalidationBase($sourceCode, $enable, $expectedValid);
+    }
+
+    /**
+     * Tests Source enabling and disabling when no Source Items are connected to current Source.
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_links.php
+     * @dataProvider IndexerInvalidationNoSourceItemsDataProvider
+     * @param string $sourceCode
+     * @param bool $enable
+     * @param bool $expectedValid
+     */
+    public function testIndexerInvalidationNoSourceItems(string $sourceCode, bool $enable, bool $expectedValid)
+    {
+        $this->indexerInvalidationBase($sourceCode, $enable, $expectedValid);
+    }
+
+    /**
+     * Tests Source enabling and disabling when both Stocks and Source Items are connected to current Source.
+     *
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_links.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     * @dataProvider IndexerInvalidationFullDataProvider
+     * @param string $sourceCode
+     * @param bool $enable
+     * @param bool $expectedValid
+     */
+    public function testIndexerInvalidationFull(string $sourceCode, bool $enable, bool $expectedValid)
+    {
+        $this->indexerInvalidationBase($sourceCode, $enable, $expectedValid);
+    }
+
+    /**
+     * Contains generic logic for all Indexer invalidation after Source enabled or disabled tests.
+     *
+     * @param string $sourceCode
+     * @param bool $enable
+     * @param bool $expectedValid
+     */
+    private function indexerInvalidationBase(string $sourceCode, bool $enable, bool $expectedValid)
```

Comment:
if it is `assert` rename method to `assertWhatIWantToCheck`

Reply:
@naydav no. We need to run the same test with the different fixtures loaded. This method contains generic logic for all those tests.